src/rust/inetstack/protocols/icmpv4/datagram/message.rs

```diff
@@ -26,7 +26,7 @@ pub struct Icmpv4Message {
     ethernet2_hdr: Ethernet2Header,
     ipv4_hdr: Ipv4Header,
     icmpv4_hdr: Icmpv4Header,
-    data: DemiBuffer,
+    data: Option<DemiBuffer>,
```

Comment:
If the code panics when `data` is `None`, then why make it an `Option`?

Reply:
So I made it an option because we need to eventually transfer ownership out of the PacketBuf structure. The panic ensures that we don't mistake this PacketBuf for one without a payload at all. It should be unreachable, so maybe I should use that instead?